tests/unit/EntityId/PrefixMappingEntityIdParserTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Tests\EntityId;
+
+use Wikibase\DataModel\Entity\BasicEntityIdParser;
+use Wikibase\DataModel\Entity\EntityIdParsingException;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Services\EntityId\PrefixMappingEntityIdParser;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterTypeException;
+
+/**
+ * @covers Wikibase\DataModel\Services\EntityId\PrefixMappingEntityIdParser
+ *
+ * @license GPL-2.0+
+ */
+class PrefMappixingEntityIdParserTest extends \PHPUnit_Framework_TestCase {
+
+	private function getRegularParser( $input, EntityId $returnValue ) {
+		$regularParser = $this->getMock( BasicEntityIdParser::class );
+		$regularParser->expects( $this->once() )
+			->method( 'parse' )
+			->with( $input )
+			->will( $this->returnValue( $returnValue ) );
+		return $regularParser;
+	}
+
+	/**
+	 * @dataProvider validIdSerializationProvider
+	 */
+	public function testGivenNoPrefixMapping_prefixIsAddedToIdSerialization( $prefix, $input, $expected ) {
+		$parser = new PrefixMappingEntityIdParser(
+			$prefix,
+			[],
+			$this->getRegularParser( $prefix . ':' . $input, $expected )
+		);
+		$this->assertEquals( $expected, $parser->parse( $input ) );
+	}
+
+	public function validIdSerializationProvider() {
+		return array(
+			array( 'wikidata', 'Q1337', new ItemId( 'wikidata:Q1337' ) ),
+			array( 'foo', 'P123', new PropertyId( 'foo:P123' ) ),
+			array( 'foo', 'bar:Q1337', new ItemId( 'foo:bar:Q1337' ) ),
+			array( 'foo', 'foo:Q1337', new ItemId( 'foo:foo:Q1337' ) ),
+		);
+	}
+
+	public function testGivenPrefixIsMapped_mappedOneIsUsed() {
+		$prefixMapping = [
+			'foo' => [ 'bar' => 'wd' ]
+		];
+		$regularParser = $this->getRegularParser( 'wd:Q1337', new ItemId( 'wd:Q1337' ) );
+		$parser = new PrefixMappingEntityIdParser( 'foo', $prefixMapping['foo'], $regularParser );
+		$this->assertEquals( new ItemId( 'wd:Q1337' ), $parser->parse( 'bar:Q1337' ) );
+	}
+
+	public function testEntityIdParsingExceptionsAreNotCaught() {
+		$regularParser = $this->getMock( BasicEntityIdParser::class );
+		$regularParser->expects( $this->once() )
+			->method( 'parse' )
+			->with( 'wikidata:QQQ' )
```

Comment:
Looking at the name of the test, I'm wondering why you are also testing that the call is made just once, and that a given parameter value is passed along. These are different concerns already tested elsewhere.